src/exec.cpp

```diff
@@ -1099,8 +1070,9 @@ void exec_job(parser_t &parser, job_t *j) {
                 }
 
                 // This is the parent process. Store away information on the child, and possibly
-                // fice it control over the terminal.
+                // give it control over the terminal.
                 p->pid = pid;
+                set_child_group(j, p->pid);
```

Comment:
We need to either set `pgrep_set = true` here or drop `pgrp_set` entirely. We can probably do the latter now.

Reply:
Good catch. I had dropped it in a later commit that wasn't part of this PR, oops.